### PR TITLE
[MRESOLVER-408] Make master 2.0.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
* make version 2.0.0-SNAPSHOT
* build time JDK requirement: raise to 11 (raise as needed, preferably use LTS versions)
* existing modules are `release=8`
* drop animal sniffer
* redefine enforceBytecodeVersion: we do produce Java 8 stuff, but for **testing** we may want and will want to use something that is 8+ (ie. Jetty 12)
* fix sisu (was invoked twice, last parent POM update)
* compiler: proactively disable annotation processing (we do not use those, but Sisu APT is present, we do not want it auto-picked up, we explicitly index using sisu mojo)
* compiler: be chatty about deprecations, to help us down the line removing them

---

https://issues.apache.org/jira/browse/MRESOLVER-408

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,18 +85,21 @@
...
+    <guiceVersion>6.0.0</guiceVersion>
     <slf4jVersion>1.7.36</slf4jVersion>
     <!-- used by supplier and demo only -->
-    <mavenVersion>3.9.4</mavenVersion>
-    <minimalMavenBuildVersion>[3.8.7,)</minimalMavenBuildVersion>
-    <minimalJavaBuildVersion>[1.8.0-362,)</minimalJavaBuildVersion>
-    <project.build.outputTimestamp>2023-09-22T18:13:46Z</project.build.outputTimestamp>
+    <mavenVersion>3.9.5</mavenVersion>
+    <minimalMavenBuildVersion>[3.8.8,)</minimalMavenBuildVersion>
+    <minimalJavaBuildVersion>[17.0.6,)</minimalJavaBuildVersion>
```
Comment (michael-o) [inline pom.xml:101]: What is the benefit to require Java 17 to build it? Why is 11 not enough?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:101]: IMHO next to last LTS is good compromise for **build time requirement** for our anyway dated stack we use. In other words, I want to encourage experiments, and am willing even to go for Java 21 but there are some spotless issues (@slawekjaranowski plz help). Also, I must emphasize: **most resolver modules will be still release=8 compiled**, but probably not all. As we may have new modules, that do want all the new nice things we can get.

Am pretty much sure, that our users run Maven on latest Java versions (unlike "aligners"), so as resolver being one of the Maven fundamentals libraries, let's make a leap here.

**Review by slawekjaranowski — COMMENTED** [review]
for me ok - please also fix in Jenkinsfile

File: pom.xml (hunk 1 above)
Comment (michael-o) [reply pom.xml:101]: I still don't see a reason to artificially limit this. You still can build with newer ones.

Comment (michael-o) [reply pom.xml:101]: Lower limit. There is no technical justification.

Comment (cstamas) [reply pom.xml:101]: There ARE technical (and human) justifications. For example, remember this thread? :smile: 
https://lists.apache.org/thread/k5sc3w7w4xkgjfzyxxltg06722dvspk7

In short, we tend to say "Java 8" but there are important **Java updates**, hence there is java 8u172 from 2018 and Java 8u371 from 2023. This thread **happened** to us, and clearly shows there is sense to set "lower limit". I just dont want to waste my time and resources on things like happened in that thread.

Comment (michael-o) [reply pom.xml:101]: You can enforce 8u300+, no?

Comment (cstamas) [reply pom.xml:101]: I can, but why would I? I want to add modules that do HTTP/2, and require Java11+, moreover, for testing those I may want to use dependencies (test scoped) that are Java17...

What would limiting build time Java requirement to 8u300 help with those above?

File: Jenkinsfile [hunk 2]
```diff
@@ -19,5 +19,5 @@
 
 // Currently we can only build with JDK8+ based on the usage
```
Comment (gnodet) [inline Jenkinsfile:20]: I think this comment now looks wrong.

**Review by gnodet — CHANGES_REQUESTED** [review]

**Review by olamy — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (michael-o) [reply pom.xml:99]: Do you have Java 17 test deps already?

**Review by gnodet — APPROVED** [review]
LGTM, though if there's no need to bump the build time JDK requirement, I would keep it low.  Whenever the needs come, we can bump it.

**Comment by jira-importer** [comment]
Resolve #1084

**Comment by jira-importer** [comment]
Resolve #1084

**Pull request merged**